src/command/restore/restore.c

```diff
@@ -2068,11 +2069,40 @@ restoreProcessQueue(const Manifest *const manifest, List **const queueList)
         MEM_CONTEXT_END();
 
         // Now put all files into the processing queues
+        const bool isFilterSet = cfgOptionTest(cfgOptFilter);
         for (unsigned int fileIdx = 0; fileIdx < manifestFileTotal(manifest); fileIdx++)
         {
             const ManifestFilePack *const filePack = manifestFilePackGet(manifest, fileIdx);
             const ManifestFile file = manifestFileUnpack(manifest, filePack);
 
+            if (isFilterSet)
+            {
+                Oid dbNode, spcNode, relNode;
+                // If this file is located in the default tablespace
+                if (sscanf(strZ(file.name), MANIFEST_TARGET_PGDATA "/" PG_PATH_BASE "/%u/%u", &dbNode, &relNode) == 2)
+                {
+                    if (!isRelationNeeded(dbNode, DEFAULTTABLESPACE_OID, relNode))
+                        continue;
+                }
+                // If this file is located in a non-built-in tablespace
+                else if (sscanf(strZ(file.name), MANIFEST_TARGET_PGTBLSPC "/%u", &spcNode) == 1)
+                {
+                    StringList *lst = strLstNewSplitZ(file.name, "/");
+                    if (strLstSize(lst) == 5)
+                    {
+                        if (
+                            sscanf(strZ(strLstGet(lst, 3)), "%u", &dbNode) == 1 &&
+                            sscanf(strZ(strLstGet(lst, 4)), "%u", &relNode) == 1 &&
```

Comment:
I don't quite understand, where is the test for these lines? I expected it to be something like 123/qwe/456/789

Reply:
https://github.com/arenadata/pgbackrest/pull/44/files#diff-6d9ab887a8ce7469b6f93279461cc94e5a2f4d43bbaaba5287b5caa94ac8ff4bR3414-R3415